pyro/ops/sumproduct.py

```diff
@@ -44,7 +44,7 @@ def memoized_sum_keepdim(tensor, dim):
     return result
 
 
-def sumproduct(factors, target_shape=(), optimize=True):
+def sumproduct(factors, target_shape=(), optimize=True, device="cpu"):
```

Comment:
What do you think about switching to default to `device=None` so that this works with `torch.set_default_tensor_type()` even when `device` is omitted?

Reply:
Yeah, good point. Will change.